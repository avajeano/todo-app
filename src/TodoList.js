import React, { useState } from "react";
import { v4 as uuid } from "uuid";
import TodoListForm from "./TodoListForm";
import Todo from "./Todo";

const TodoList = () => {
    const INITIAL_STATE = [{id: uuid(), todo: ''}]
    const [todos, setTodos] = useState(INITIAL_STATE);
    const addTodo = (newTodo) => {
        setTodos(todos => [...todos, {...newTodo, id:uuid()}])
    }
    const removeTodo = (id) => {
        setTodos(todos => todos.filter(todo => todo.id !== id));
    }
    return (
        <div>
            <h2>Todo List</h2>
            <TodoListForm addTodo={addTodo} />
            <div>
                {todos.map(todo => (
                    <Todo key={todo.id} id={todo.id} todo={todo.todo} removeTodo={removeTodo}/>
                ))}
            </div>
        </div>
    )
}

export default TodoList;