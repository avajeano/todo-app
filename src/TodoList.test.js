import React from "react";
import { render, fireEvent } from '@testing-library/react';
import TodoList from "./TodoList";

// Todo list smoke test 
it('renders without crashing', () => {
    render(<TodoList />)
})

// Todo list snapshot test
it('matches snapshot', () => {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot()
})

// test that remove button works 
it('removes the todo when the x is clicked', () => {
    const { getByLabelText, getByText, queryByText } = render(<TodoList />);

    // add a todo
    fireEvent.change(getByLabelText("Todo"), {target: { value: "test todo" } });

    //ensure the todo is added 
    expect(queryByText("X")).toBeInTheDocument();

    // click the remove button
    fireEvent.click(getByText("X"));

    // ensure the todo is removed
    expect(queryByText("X")).not.toBeInTheDocument();
})