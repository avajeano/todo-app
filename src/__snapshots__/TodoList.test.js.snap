// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Todo List
    </h2>
    <form>
      <label
        for="todo"
      >
        Todo
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="Todo"
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
    <div>
      <ul>
        <li>
          <button>
            X
          </button>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
